src/machine/machine_avr.go

```diff
@@ -142,3 +143,77 @@ func (a ADC) Get() uint16 {
 
 	return uint16(avr.ADCL.Get()) | uint16(avr.ADCH.Get())<<8
 }
+
+var Ticks int64     // nanoseconds since start
+var tickNanos int64 // nanoseconds per each tick
+
+func InitMonotonicTimer() {
+	tickNanos = 0
+	Ticks = 0
+
+	interrupt.New(avr.IRQ_TIMER0_OVF, func(i interrupt.Interrupt) {
+		Ticks += tickNanos
+	})
+
+	// initial initialization of the Timer0
+	// - Mask interrupt
+	avr.TIMSK0.ClearBits(avr.TIMSK0_TOIE0 | avr.TIMSK0_OCIE0A | avr.TIMSK0_OCIE0B)
+
+	// - Write new values to TCNT2, OCR2x, and TCCR2x.
+	avr.TCNT0.Set(0)
+	avr.OCR0A.Set(0xff)
+	// - Set mode 3
+	avr.TCCR0A.Set(avr.TCCR0A_WGM00 | avr.TCCR0A_WGM01)
+	// - Set prescaler 1
+	avr.TCCR0B.Set(avr.TCCR0B_CS00)
+
+	adjustMonotonicTimer()
+
+	// - Unmask interrupt
+	avr.TIMSK0.SetBits(avr.TIMSK0_TOIE0)
+}
+
+func adjustMonotonicTimer() {
+	// adjust the tickNanos
+	tickNanos = currentTickNanos()
+	println("adjustMonotonicTimer tickNanos=", tickNanos, "ticks=", Ticks)
```

Comment:
Remove println if it's for debugging during development of this PR.

Reply:
removed